core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -121,6 +121,9 @@ public boolean isAccessible(Map context, Object target, Member member, String pr
     protected boolean checkStaticMethodAccess(Member member) {
         int modifiers = member.getModifiers();
         if (Modifier.isStatic(modifiers)) {
+            if (member instanceof Field) {
+                return false;  // Refuse access to all static fields (even if allowStaticMethodAccess true)
+            }
```

Comment:
but access to public static fields were allowed even before

Reply:
Hello @lukaszlenart .

Access to public static fields is still allowed (the unit tests for this PR check that), but it works again due to the OGNL 3.1.22 fix (which restored the 3.1.18 public static field access behaviour - no isAccessible check for public static fields).
Basically since the OGNL 3.1.x "public static field access behaviour contract" means checkStaticMethodAccess should never be called in the flow for public static fields, the above was just to block all field access. 

But now that you draw attention to it again, it would probably be clearer/better to base it on the public state, even if the check "never should happen".  I'll update the PR to reflect that.

Reply:
Hello.

This PR has been updated in response to the comment above (so the method will return true for static public fields, even if that path is not normally called with most OGNL 3.1.x versions).  The `SecurityMemberAccess` module `checkStaticMethodAccess` method was deprecated and replaced with the more appropriately named `checkStaticMemberAccess` (so that in the unlikely event someone extended the class things should still work OK).

One additional cleanup was added as well (noticed during the similar PR for 2.6.x), to utilize a constant instead of a literal for the static access injection name in OgnlValueStackFactory.